src/com/seafile/seadroid2/cameraupload/CameraUploadConfigActivity.java

```diff
@@ -0,0 +1,247 @@
+                        toRemoveList.add(dir);
+                }
+            }
+            CameraUploadDBHelper.getInstance().removeDirList(toRemoveList);
+
+            Iterator iterator = map.entrySet().iterator();
+            while (iterator.hasNext()) {
+                Map.Entry<String, Boolean> pair = (Map.Entry) iterator.next();
+                summaryPaths.add(pair.getKey());
+                // insert or update new data
+                CameraUploadDBHelper.getInstance().saveSelectedDirectory(pair.getKey(), pair.getValue());
+                iterator.remove();
+            }
+
+            if (!summaryPaths.isEmpty())
+                sm.saveLocalDirPath(summaryPaths.toString());
+        }
+
+        Intent intent = new Intent();
+        // update cloud library data
+        if (mSeafRepo != null && mAccount != null) {
+            intent.putExtra(SeafilePathChooserActivity.DATA_REPO_NAME, mSeafRepo.name);
+            intent.putExtra(SeafilePathChooserActivity.DATA_REPO_ID, mSeafRepo.id);
+            intent.putExtra(SeafilePathChooserActivity.DATA_DIR, mSeafRepo.root);
+            intent.putExtra(SeafilePathChooserActivity.DATA_ACCOUNT, mAccount);
+        }
+
+        if (!summaryPaths.isEmpty()) {
+            intent.putExtra(SeafilePathChooserActivity.DATA_DIRECTORY_PATH, summaryPaths.toString());
+        }
+
+        setResult(RESULT_OK, intent);
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (mCurrentPosition == 0) {
+            setResult(RESULT_CANCELED);
+            super.onBackPressed();
+        } else {
+            // navigate to previous page when press back button
+            mCurrentPosition -= 1;
+            mIndicator.setCurrentItem(mCurrentPosition);
+        }
+    }
+
+    public boolean isChooseLibPage() {
+        return isChooseLibPage;
+    }
+
+    public boolean isChooseDirPage() {
+        return isChooseDirPage;
+    }
+
+    public void saveDataPlanAllowed(boolean isAllowed) {
+        sm.saveDataPlanAllowed(isAllowed);
+    }
+
+    public void saveVideosAllowed(boolean isAllowed) {
+        sm.saveVideosAllowed(isAllowed);
+    }
+
+    public void saveCustomScanDir(boolean isCustom) {
+        sm.saveCustomScanDir(isCustom);
+    }
+
+    public boolean isCustomScanDir() {
+        return sm.isCustomScanDir();
+    }
+
+    class CameraUploadConfigAdapter extends FragmentStatePagerAdapter {
+
+        public CameraUploadConfigAdapter(FragmentManager fm) {
+            super(fm);
+        }
+
+        // This method controls which fragment should be shown on a specific screen.
+        @Override
+        public Fragment getItem(int position) {
+
+            if (isChooseLibPage) {
+                switch (position) {
+                    case 0:
+                        mCloudLibFragment = new CloudLibraryFragment();
+                        return mCloudLibFragment;
+                    default:
+                        return null;
+                }
```

Comment:
The above ~6 lines inside the `if` block can be simplified as `return position == 0 ? new CloudLibraryFragment() : null`